src/net_processing.cpp

```diff
@@ -3786,7 +3792,7 @@ bool PeerLogicValidation::SendMessages(CNode* pto)
             pto->vInventoryBlockToSend.clear();
 
             // Check whether periodic sends should happen
-            bool fSendTrickle = pto->fWhitelisted;
+            bool fSendTrickle = pto->HasPermission(PF_NOBAN);
```

Comment:
Why is this `noban` and not `relay`?

Reply:
To keep backward compatibility. 

Before, if the peer was whitelisted, it did not matter whether he was also in `whitelistrelay`.

Reply:
But `relay` is also set to true for backward compatibility, no?

Reply:
only if the peer is also in `-whitelistrelay`

Reply:
Which is enabled by default for exactly that reason, no?

Reply:
Well, regardless, the old code does not look `-whitelistrelay` is 1 or 0 to determine `fSendTrickle`. 

This would be a breaking change to make it depends on it.
I am quite neutral to that, I am unsure what `fSendTrickle` is doing, but I think for this PR it should preserve the old behavior.

Another alternative is to have a specific permission just for that and turn it on if no specific permission are set regardless of `-whitelistrelay`

Reply:
I just encountered this piece of code and found this to be pretty confusing.  It seems to me that the behavior of relaying transactions immediately without waiting for our poisson timers to fire should be its own "net permission", and not piggy-backed onto NOBAN.

Reply:
According to @NicolasDorier it is needed for backward compatibility. However, it should still be possible to split up into its own permission while retaining backward compatibility.

Reply:
If we want to spilt this permission, it's best to do that before 0.19 is final.

Reply:
Splitting it would be a new feature, so it might be a bit late for 0.19